### PR TITLE
feature/private cloud platform config

### PR DESCRIPTION
**Pull request by JohannesRudolph**
- fix: update the OpenStack integration instructions
- fix: update the Cloud Foundry integration instructions

### CODE REVIEW

File: docs/meshstack.cloudfoundry.index.md [hunk 1]
```diff
@@ -3,62 +3,49 @@ id: meshstack.cloudfoundry.index
...
 
-For Cloud Foundry, meshStack provides org and space creation and configuration, user management and SSO via Cloud Foundry's UAA.
+meshStack supports project creation, configuration, access control, quota management and billing for Cloud Foundry.
 
 ## Integration Overview
 
-> The recommended way to set up Cloud Foundry as a meshPlatform is via the public terraform [Cloud Foundry meshPlatform Module](https://github.com/meshcloud/terraform-cloudfoundry-meshplatform).
-
-To enable integration with Cloud Foundry, Platform Operators deploy and configure the meshStack Cloud Foundry connector to make Cloud Foundry platforms available at their meshStack instance.
+To enable integration with Cloud Foundry, Platform Operators configure one or multiple `meshPlatform`s of `PlatformType` Cloud Foundry in the [Platform Administration](./administration.platforms.md) in meshPanel.
```
Comment (Jelledb) [inline docs/meshstack.cloudfoundry.index.md:12]:
```suggestion
To enable integration with Cloud Foundry, Platform Operators configure one or multiple `Platform`s of `PlatformType` Cloud Foundry in the [Platform Administration](./administration.platforms.md) in meshPanel.
```

File: docs/meshstack.cloudfoundry.index.md [hunk 2]
```diff
@@ -3,62 +3,49 @@ id: meshstack.cloudfoundry.index
...
+We have officially validated meshStack with support for Cloud Foundry `v3.108.0`.
 
-The tenant [replication](./meshcloud.tenant.md) ensures spaces and orgs are created within the CF platform and appropriate permission rights are set when users access the CF platform. If a user's meshProject permissions are modified, meshStack updates the permissions for this user accordingly within the CF platform.
+### UAA Federated Users
 
-![Cloud Foundry Architecture](assets/cf-architecture.svg)
+meshStack will identify and assign users to roles in Cloud Foundry based on their `euid` (external user id) as described in [Identity Federation](meshstack.identity-federation.md#externally-provisioned-identities).
 
-## Cloud Foundry Versions
+You should set up your Cloud Foundry and UAA UAA so that they store the `euid` value in the `User.userName` field of the [UAA User object](https://docs.cloudfoundry.org/api/uaa/version/77.8.0/index.html#get) and `User.username` field of the [Cloud Foundry user object](http://v3-apidocs.cloudfoundry.org/version/3.164.0/index.html#users).
```
Comment (Jelledb) [inline docs/meshstack.cloudfoundry.index.md:23]: Just checking whether you wanted to mention UAA UAA twice.

File: docs/meshstack.cloudfoundry.index.md [hunk 3]
```diff
@@ -3,62 +3,49 @@ id: meshstack.cloudfoundry.index
...
 
-meshcloud officially validates compatibility with the following API versions:
+## Integration Configuration
 
-* `2.155.0`
-* `2.181.0`
+> The recommended way to set up Cloud Foundry as a meshPlatform is via the public terraform [Cloud Foundry meshPlatform Module](https://github.com/meshcloud/terraform-cloudfoundry-meshplatform).
```
Comment (Jelledb) [inline docs/meshstack.cloudfoundry.index.md:27]:
```suggestion
> The recommended way to integrate Cloud Foundry as a platform in meshStack is via the public terraform [Cloud Foundry Platform Module](https://github.com/meshcloud/terraform-cloudfoundry-meshplatform).
```

Hmm now we did call the repo meshPlatform in the end. I leave it up to you whether you want to keep the meshPlatform naming but we did kind of decide to get rid of the meshNaming in our product.

File: docs/meshstack.cloudfoundry.index.md [hunk 4]
```diff
@@ -3,62 +3,49 @@ id: meshstack.cloudfoundry.index
...
+meshStack requires two technical admin user accounts for integration with Cloud Foundry. We recommend your assign these 
+users the following [Cloud Foundry roles](https://docs.cloudfoundry.org/concepts/roles.html) and [UAA scopes](https://docs.cloudfoundry.org/concepts/architecture/uaa.html#scopes).
 
-## Cloud Foundry Access Workflow
+- the `replicator` admin account must have permission to create orgs, spaces, groups and assign roles. This user requires the
+  `Admin` role and additionally needs the `cloud_controller.admin`, `uaa.admin` and  `scim.read` scopes in UAA.
+- the `metering` admin account is used to collect [metering data](./meshstack.cloudfoundry.metering.md). This requires the `Global Auditor`.
 
-The full workflow to access the Cloud Foundry platform is as follows:
+The tenant [replication](./meshcloud.tenant.md) ensures spaces and orgs are created within the CF platform and appropriate permission rights are set when users access the CF platform. If a user's meshProject permissions are modified, meshStack updates the permissions for this user accordingly within the CF platform.
```
Comment (Jelledb) [inline docs/meshstack.cloudfoundry.index.md:38]:
```suggestion
The tenant [replication](./meshcloud.tenant.md) ensures spaces and orgs are created within the CF platform and appropriate permission rights are set when users access the CF platform. If a user's project permissions are modified, meshStack updates the permissions for this user accordingly within the CF platform.
```

File: docs/meshstack.openstack.index.md [hunk 5]
```diff
@@ -8,21 +8,18 @@ meshStack supports project creation, configuration, access control, quota manage
 
 ## Integration Overview
 
-> Self-service OpenStack platform configuration is currently not available in the [Platform Administration](./administration.platforms.md) in meshPanel.
-> Please request assistance from our customer success team to configure your OpenStack platform.
+To enable integration with OpenStack, Platform Operators configure one or multiple `meshPlatform`s of `PlatformType` OpenStack in the [Platform Administration](./administration.platforms.md) in meshPanel.
```
Comment (Jelledb) [inline docs/meshstack.openstack.index.md:11]:
```suggestion
To enable integration with OpenStack, Platform Operators configure one or multiple `Platform`s of `PlatformType` OpenStack in the [Platform Administration](./administration.platforms.md) in meshPanel.
```

File: docs/meshstack.openstack.index.md [hunk 6]
```diff
@@ -43,37 +40,31 @@ meshStack also supports the following OpenStack APIs for advanced features
...
 During replication, meshStack will make sure that users have access to the OpenStack projects they are assigned to in meshStack.
-Users will be assigned to the according groups (per meshProject) that meshStack creates in OpenStack.
-If meshStack is about to assign a user that does not exist in OpenStack yet, meshStack will create this user with the
-according IDP attributes and create the user in the Keystone domain `MeshUsers`.
+If meshStack finds that a user does not exist in OpenStack yet, meshStack will log a replication remark and skip assigning that user's permission. 
+When the user is created in OpenStack at any later time, meshStack will pick up this user to assign him to the according groups in the next replication.
 
-That way once the user logs in via the IdP, he will be mapped to the user created by meshStack.
-If the user already exists in OpenStack during replication, meshStack will pick up this user to assign him to the according groups.
+Users will be assigned to the according groups (per meshProject) that meshStack creates in OpenStack.
```
Comment (Jelledb) [inline docs/meshstack.openstack.index.md:65]:
```suggestion
Users will be assigned to the according groups (per project in meshStack) that meshStack creates in OpenStack.
```
I think if you do really want to differentiate between a OS project and a meshStack project this is a good pattern.

**Review by Jelledb — APPROVED** [review]

File: docs/meshstack.cloudfoundry.index.md (hunk 3 above)
Comment (JohannesRudolph) [reply docs/meshstack.cloudfoundry.index.md:27]: @Jelledb all of the platform integration tf modules are called `terraform-$provider-meshplatform`. The prefix `terraform-$provider` is mandatory by terraform registry conventions. If you want to call them differently, we'd need to migrate/rename all of them. If you want to get rid of that naming everywhere, that would require a dedicated backlog item.

File: docs/meshstack.openstack.index.md (hunk 6 above)
Comment (JohannesRudolph) [reply docs/meshstack.openstack.index.md:65]: Yeah, dropping the mesh* prefix requires becoming a bit more verbose in those docs.

**Pull request merged**